lib/cluster.js

```diff
@@ -293,28 +293,47 @@ function masterInit() {
     cluster.emit('setup', settings);
   }
 
-  var debugPortOffset = 1;
+  var debugPortOffset = 0;
+
+  // match[1]: debug argument (e.g. --debug)
+  // match[2]: host part in debug socket (e.g. [::] or 127.0.0.1)
+  // match[3]: debug port (e.g. 5858)
+  const debugArgRegex =
+    /^(--inspect|--debug|--debug-(?:brk|port))(?:=(.+:)?(\d+))?$/;
+
+  for (var i = 0; i < process.execArgv.length; i++) {
+    if (process.execArgv[i].match(debugArgRegex)) {
+      debugPortOffset = 1;
+      break;
+    }
+  }
 
   function createWorkerProcess(id, env) {
     var workerEnv = util._extend({}, process.env);
     var execArgv = cluster.settings.execArgv.slice();
     var debugPort = 0;
+    var debugHost = '';
 
     workerEnv = util._extend(workerEnv, env);
     workerEnv.NODE_UNIQUE_ID = '' + id;
 
     for (var i = 0; i < execArgv.length; i++) {
-      var match = execArgv[i].match(
-        /^(--inspect|--debug|--debug-(brk|port))(=\d+)?$/
-      );
+      var match = execArgv[i].match(debugArgRegex);
 
       if (match) {
         if (debugPort === 0) {
-          debugPort = process.debugPort + debugPortOffset;
+          if (match[3]) {
+            debugPort = +match[3] + debugPortOffset;
+            if (match[2]) {
```

Comment:
This should be outside the `match[3]` block, right? The port is also optional, so the user might just give the hostname?